# Подготовка файлов и программ

In [ ]:
pip install -q condacolab

In [ ]:
import condacolab

In [ ]:
condacolab.install()

⏬ Downloading https://github.com/jaimergp/miniforge/releases/latest/download/Mambaforge-colab-Linux-x86_64.sh...
📦 Installing...
📌 Adjusting configuration...
🩹 Patching environment...
⏲ Done in 0:00:38
🔁 Restarting kernel...


In [ ]:
!conda install -c bioconda seqtk

Solving environment: / - \ | / - \ done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - seqtk


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2021.10.8  |       ha878542_0         139 KB  conda-forge
    certifi-2021.10.8          |   py37h89c1867_1         145 KB  conda-forge
    conda-4.11.0               |   py37h89c1867_1         983 KB  conda-forge
    seqtk-1.3                  |       h5bf99c6_3          38 KB  bioconda
    ------------------------------------------------------------
                                           Total:         1.3 MB

The following NEW packages will be INSTALLED:

  seqtk              bioconda/linux-64::seqtk-1.3-h5bf99c6_3

The following packages will be UPDATED:

  ca-certificates                      2020.12.5-ha878542_0 --> 2021.10.8-ha878542_0
  certifi                      

In [ ]:
!conda install -c bioconda trimmomatic

Solving environment: - \ | / - \ | / - \ | done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - trimmomatic


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    openjdk-8.0.312            |       h7f98852_0        97.6 MB  conda-forge
    python_abi-3.7             |          2_cp37m           4 KB  conda-forge
    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda
    ------------------------------------------------------------
                                           Total:        97.8 MB

The following NEW packages will be INSTALLED:

  openjdk            conda-forge/linux-64::openjdk-8.0.312-h7f98852_0
  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2

The following packages will be UPDATED:

  python_abi                                    3.7-1_cp37m --> 3.7-2_cp37m



trimmomatic-0.39  

In [ ]:
!pip install macs2 intervene

     |████████████████████████████████| 2.1 MB 14.0 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
     |████████████████████████████████| 5.5 MB 29.4 MB/s 
     |████████████████████████████████| 12.5 MB 35.6 MB/s 
     |████████████████████████████████| 11.2 MB 48.2 MB/s 
     |████████████████████████████████| 11.3 MB 46.2 MB/s 
  Using cached numpy-1.21.5-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)
     |████████████████████████████████| 38.1 MB 1.2 MB/s 
     |████████████████████████████████| 292 kB 70.8 MB/s 
     |████████████████████████████████| 40 kB 5.3 MB/s 
     |████████████████████████████████| 98 kB 8.2 MB/s 
     |████████████████████████████████| 247 kB 72.3 MB/s 
     |████████████████████████████████| 895 kB 64.5 MB/s 
     |████████████████████████████████| 1.1 MB 66.9 MB/s 
     |████████████████████████████████| 4.3 MB 37.7 MB/s 
     |███████████████████

In [ ]:
!apt-get install bedtools

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  bedtools
0 upgraded, 1 newly installed, 0 to remove and 39 not upgraded.
Need to get 577 kB of archives.
After this operation, 2,040 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]
Fetched 577 kB in 0s (6,377 kB/s)
Selecting previously unselected package bedtools.
(Reading database ... 155332 files and directories currently installed.)
Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...
Unpacking bedtools (2.26.0+dfsg-5) ...
Setting up bedtools (2.26.0+dfsg-5) ...


In [ ]:
!apt-get install bowtie2 samtools

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  cwltool libc-ares2 libhts2 libjs-bootstrap nodejs nodejs-doc
  python-asn1crypto python-avro python-cachecontrol python-certifi
  python-cffi-backend python-chardet python-cryptography python-enum34
  python-html5lib python-idna python-ipaddress python-isodate python-lockfile
  python-mistune python-openssl python-pkg-resources python-pyparsing
  python-rdflib python-rdflib-jsonld python-requests python-ruamel.yaml
  python-schema-salad python-shellescape python-six python-sparqlwrapper
  python-typing python-urllib3 python-webencodings
Suggested packages:
  bowtie2-examples python-cryptography-doc python-cryptography-vectors
  python-enum34-doc python-genshi python-lxml python-lockfile-doc
  python-openssl-doc python-openssl-dbg python-setuptools python-pyparsing-doc
  python-rdflib-doc python-rdflib-tools python-socks python-ntlm
The f

In [ ]:
!conda install -c bioconda bowtie2

Solving environment: - \ | / - \ | / - \ | / done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - bowtie2


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    bowtie2-2.3.5.1            |   py37he513fc3_0        12.0 MB  bioconda
    perl-5.32.1                | 0_h7f98852_perl5        14.5 MB  conda-forge
    tbb-2020.2                 |       h4bd325d_4         1.5 MB  conda-forge
    ------------------------------------------------------------
                                           Total:        27.9 MB

The following NEW packages will be INSTALLED:

  bowtie2            bioconda/linux-64::bowtie2-2.3.5.1-py37he513fc3_0
  perl               conda-forge/linux-64::perl-5.32.1-0_h7f98852_perl5
  tbb                conda-forge/linux-64::tbb-2020.2-h4bd325d_4



tbb-2020.2           | 1.5 MB    | : 100% 1.0/1 [00:00<00:00,  2

## Установка FastQC

In [ ]:
!java -version

openjdk version "1.8.0_312"
OpenJDK Runtime Environment (Zulu 8.58.0.13-CA-linux64) (build 1.8.0_312-b07)
OpenJDK 64-Bit Server VM (Zulu 8.58.0.13-CA-linux64) (build 25.312-b07, mixed mode)


In [ ]:
!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip

--2022-03-09 16:51:23--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip
Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4
Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 10249221 (9.8M) [application/zip]
Saving to: ‘fastqc_v0.11.9.zip’

fastqc_v0.11.9.zip  100%[===================>]   9.77M  20.5MB/s    in 0.5s    

2022-03-09 16:51:23 (20.5 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]



In [ ]:
!unzip fastqc_v0.11.9.zip

Archive:  fastqc_v0.11.9.zip
  inflating: FastQC/cisd-jhdf5.jar   
   creating: FastQC/Configuration/
  inflating: FastQC/Configuration/adapter_list.txt  
  inflating: FastQC/Configuration/contaminant_list.txt  
  inflating: FastQC/Configuration/limits.txt  
  inflating: FastQC/fastqc           
  inflating: FastQC/fastqc_icon.ico  
   creating: FastQC/Help/
   creating: FastQC/Help/1 Introduction/
   creating: FastQC/Help/1 Introduction/.svn/
  inflating: FastQC/Help/1 Introduction/.svn/entries  
   creating: FastQC/Help/1 Introduction/.svn/props/
   creating: FastQC/Help/1 Introduction/.svn/text-base/
  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  
   creating: FastQC/Help/1 Introduction/.svn/tmp/
   creating: FastQC/Help/1 Introduction/.svn/tmp/props/
  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  
   creating: FastQC/Help/2 Basic Operations/
   creating: FastQC/Help/2 Basic Operations/.svn/
  inflating: FastQC/Help/2 Basic

In [ ]:
!chmod a+x FastQC/fastqc

In [ ]:
!./FastQC/fastqc --help


            FastQC - A high throughput sequence QC analysis tool

SYNOPSIS

	fastqc seqfile1 seqfile2 .. seqfileN

    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] 
           [-c contaminant file] seqfile1 .. seqfileN

DESCRIPTION

    FastQC reads a set of sequence files and produces from each one a quality
    control report consisting of a number of different modules, each one of 
    which will help to identify a different potential type of problem in your
    data.
    
    If no files to process are specified on the command line then the program
    will start as an interactive graphical application.  If files are provided
    on the command line then the program will run with no user interaction
    required.  In this mode it is suitable for inclusion into a standardised
    analysis pipeline.
    
    The options for the program as as follows:
    
    -h --help       Print this help file and exit
    
    -v --version    Print the version of the program and exit

# Выравнивание

## Скачиваем чтения



In [ ]:
# ChIP-seq на гистоновой метке
!wget https://www.encodeproject.org/files/ENCFF834DXB/@@download/ENCFF834DXB.fastq.gz
!wget https://www.encodeproject.org/files/ENCFF983KWQ/@@download/ENCFF983KWQ.fastq.gz
# ChIP-seq контроль
!wget https://www.encodeproject.org/files/ENCFF902MPL/@@download/ENCFF902MPL.fastq.gz

--2022-03-09 16:51:24--  https://www.encodeproject.org/files/ENCFF834DXB/@@download/ENCFF834DXB.fastq.gz
Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144
Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.
HTTP request sent, awaiting response... 307 Temporary Redirect
Location: https://encode-public.s3.amazonaws.com/2016/10/22/816ea1ba-a5d5-40b8-ac34-2751530e6581/ENCFF834DXB.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF834DXB.fastq.gz&AWSAccessKeyId=ASIATGZNGCNXX43YXH7V&Signature=gtp3oG%2FsfayFj%2BRDqIZqYT4JncA%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHEaCXVzLXdlc3QtMiJHMEUCIQD99u9kQ%2F1HP3YVPSEvtmigqpj81eESqkg1%2B7jl8Ds88gIgGoJNB7q1ds%2FxzQc1fD6xh6fQttvz4rWy%2FO7n9ENNRN4qgwQI2v%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDGGppTWwXlmuGHof8SrXA10ym2WJhykTPxwxwPSQUM2YT9pS6isJdXkn%2Fm9EB7V%2FhoLsfZc8j4WRx%2Bl5ndp67KXRa122dhr4Os3uluc%2FIdEUPNlWEl71296b05gWaQUy8UHYe5FKoM7oSwIDiyeXzxkRr1NX9oa

In [ ]:
# распаковка ахрива
!gzip -d /content/ENCFF834DXB.fastq.gz /content/ENCFF983KWQ.fastq.gz /content/ENCFF902MPL.fastq.gz

## FastQC (или multiQC может быть использован в работе)

In [ ]:
!./FastQC/fastqc /content/ENCFF834DXB.fastq
!./FastQC/fastqc /content/ENCFF983KWQ.fastq
!./FastQC/fastqc /content/ENCFF902MPL.fastq

Started analysis of ENCFF834DXB.fastq
Approx 5% complete for ENCFF834DXB.fastq
Approx 10% complete for ENCFF834DXB.fastq
Approx 15% complete for ENCFF834DXB.fastq
Approx 20% complete for ENCFF834DXB.fastq
Approx 25% complete for ENCFF834DXB.fastq
Approx 30% complete for ENCFF834DXB.fastq
Approx 35% complete for ENCFF834DXB.fastq
Approx 40% complete for ENCFF834DXB.fastq
Approx 45% complete for ENCFF834DXB.fastq
Approx 50% complete for ENCFF834DXB.fastq
Approx 55% complete for ENCFF834DXB.fastq
Approx 60% complete for ENCFF834DXB.fastq
Approx 65% complete for ENCFF834DXB.fastq
Approx 70% complete for ENCFF834DXB.fastq
Approx 75% complete for ENCFF834DXB.fastq
Approx 80% complete for ENCFF834DXB.fastq
Approx 85% complete for ENCFF834DXB.fastq
Approx 90% complete for ENCFF834DXB.fastq
Approx 95% complete for ENCFF834DXB.fastq
Analysis complete for ENCFF834DXB.fastq
Started analysis of ENCFF983KWQ.fastq
Approx 5% complete for ENCFF983KWQ.fastq
Approx 10% complete for ENCFF983KWQ.fastq
Appr

Далее необходимо проанализировать выдачу fastqc, произвести фильтрацию или подрезание чтений в случае необходимости. Для примера указано использование trimmomatic, также можно использовать seqtk.

Видим из выдачи fastqc, что качество первой реплики и контроля не самое лучшее - постараемся подрезать чтения, чтобы его улучшить

In [ ]:
!trimmomatic SE -phred33 ENCFF834DXB.fastq ENCFF834DXB_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \
  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36

TrimmomaticSE: Started with arguments:
 -phred33 ENCFF834DXB.fastq ENCFF834DXB_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
Automatically using 2 threads
java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)
	at org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)
	at org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)
	at org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)
	at org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)
	at org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)
	at org.u

In [ ]:
!./FastQC/fastqc ENCFF834DXB_trimmed.fastq

Started analysis of ENCFF834DXB_trimmed.fastq
Approx 5% complete for ENCFF834DXB_trimmed.fastq
Approx 10% complete for ENCFF834DXB_trimmed.fastq
Approx 15% complete for ENCFF834DXB_trimmed.fastq
Approx 20% complete for ENCFF834DXB_trimmed.fastq
Approx 25% complete for ENCFF834DXB_trimmed.fastq
Approx 30% complete for ENCFF834DXB_trimmed.fastq
Approx 35% complete for ENCFF834DXB_trimmed.fastq
Approx 40% complete for ENCFF834DXB_trimmed.fastq
Approx 45% complete for ENCFF834DXB_trimmed.fastq
Approx 50% complete for ENCFF834DXB_trimmed.fastq
Approx 55% complete for ENCFF834DXB_trimmed.fastq
Approx 60% complete for ENCFF834DXB_trimmed.fastq
Approx 65% complete for ENCFF834DXB_trimmed.fastq
Approx 70% complete for ENCFF834DXB_trimmed.fastq
Approx 75% complete for ENCFF834DXB_trimmed.fastq
Approx 80% complete for ENCFF834DXB_trimmed.fastq
Approx 85% complete for ENCFF834DXB_trimmed.fastq
Approx 90% complete for ENCFF834DXB_trimmed.fastq
Approx 95% complete for ENCFF834DXB_trimmed.fastq
Analy

In [ ]:
!trimmomatic SE -phred33 ENCFF902MPL.fastq ENCFF902MPL_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \
  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36

TrimmomaticSE: Started with arguments:
 -phred33 ENCFF902MPL.fastq ENCFF902MPL_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
Automatically using 2 threads
java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)
	at org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)
	at org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)
	at org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)
	at org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)
	at org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)
	at org.u

In [ ]:
!./FastQC/fastqc ENCFF902MPL_trimmed.fastq

Started analysis of ENCFF902MPL_trimmed.fastq
Approx 5% complete for ENCFF902MPL_trimmed.fastq
Approx 10% complete for ENCFF902MPL_trimmed.fastq
Approx 15% complete for ENCFF902MPL_trimmed.fastq
Approx 20% complete for ENCFF902MPL_trimmed.fastq
Approx 25% complete for ENCFF902MPL_trimmed.fastq
Approx 30% complete for ENCFF902MPL_trimmed.fastq
Approx 35% complete for ENCFF902MPL_trimmed.fastq
Approx 40% complete for ENCFF902MPL_trimmed.fastq
Approx 45% complete for ENCFF902MPL_trimmed.fastq
Approx 50% complete for ENCFF902MPL_trimmed.fastq
Approx 55% complete for ENCFF902MPL_trimmed.fastq
Approx 60% complete for ENCFF902MPL_trimmed.fastq
Approx 65% complete for ENCFF902MPL_trimmed.fastq
Approx 70% complete for ENCFF902MPL_trimmed.fastq
Approx 75% complete for ENCFF902MPL_trimmed.fastq
Approx 80% complete for ENCFF902MPL_trimmed.fastq
Approx 85% complete for ENCFF902MPL_trimmed.fastq
Approx 90% complete for ENCFF902MPL_trimmed.fastq
Approx 95% complete for ENCFF902MPL_trimmed.fastq
Analy

После подрезания чтений качество немного улучшилось


## Выравнивание на хромосому

Следует выбрать одну хромосому, потому что ресурсы google colab ограничены. В случае проблем, таких как Segmentation fault, error 139, следует выбрать хромосому меньшего размера.

Последовательность нуклеотидов хромосомы можно скачать по адресу https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/

Была выбрана 21-ая хромосома


In [ ]:
!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr21.fa.gz
!gzip -d chr21.fa.gz

--2022-03-09 17:13:23--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr21.fa.gz
Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163
Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 12709705 (12M) [application/x-gzip]
Saving to: ‘chr21.fa.gz’

chr21.fa.gz         100%[===================>]  12.12M  7.18MB/s    in 1.7s    

2022-03-09 17:13:26 (7.18 MB/s) - ‘chr21.fa.gz’ saved [12709705/12709705]



In [ ]:
%%time
!bowtie2-build chr21.fa chromosome_index

Settings:
  Output files: "chromosome_index.*.bt2"
  Line rate: 6 (line is 64 bytes)
  Lines per side: 1 (side is 64 bytes)
  Offset rate: 4 (one in 16)
  FTable chars: 10
  Strings: unpacked
  Max bucket size: default
  Max bucket size, sqrt multiplier: default
  Max bucket size, len divisor: 4
  Difference-cover sample period: 1024
  Endianness: little
  Actual local endianness: little
  Sanity checking: disabled
  Assertions: disabled
  Random seed: 0
  Sizeofs: void*:8, int:4, long:8, size_t:8
Input files DNA, FASTA:
  chr21.fa
Building a SMALL index
Reading reference sizes
  Time reading reference sizes: 00:00:00
Calculating joined length
Writing header
Reserving space for joined string
Joining reference sequences
  Time to join reference sequences: 00:00:00
bmax according to bmaxDivN setting: 10022154
Using parameters --bmax 7516616 --dcv 1024
  Doing ahead-of-time memory usage test
  Passed!  Constructing with these parameters: --bmax 7516616 --dcv 1024
Constructing suffix-array

In [ ]:
!mkdir bowtie2_res

In [ ]:
%%time
!bowtie2  -p 2 \
          -x chromosome_index \
          -U ENCFF834DXB_trimmed.fastq \
          -S bowtie2_res/res_DXB.sam

17283292 reads; of these:
  17283292 (100.00%) were unpaired; of these:
    15393136 (89.06%) aligned 0 times
    417150 (2.41%) aligned exactly 1 time
    1473006 (8.52%) aligned >1 times
10.94% overall alignment rate
CPU times: user 2.47 s, sys: 318 ms, total: 2.79 s
Wall time: 5min 24s


In [ ]:
%%time
!bowtie2  -p 2 \
          -x chromosome_index \
          -U ENCFF983KWQ.fastq \
          -S bowtie2_res/res_KWQ.sam

40166682 reads; of these:
  40166682 (100.00%) were unpaired; of these:
    35422229 (88.19%) aligned 0 times
    1085078 (2.70%) aligned exactly 1 time
    3659375 (9.11%) aligned >1 times
11.81% overall alignment rate
CPU times: user 5.84 s, sys: 738 ms, total: 6.58 s
Wall time: 12min 57s


In [ ]:
%%time
!bowtie2  -p 2 \
          -x chromosome_index \
          -U ENCFF902MPL_trimmed.fastq \
          -S bowtie2_res/res_MPL.sam

37572200 reads; of these:
  37572200 (100.00%) were unpaired; of these:
    31572119 (84.03%) aligned 0 times
    1187983 (3.16%) aligned exactly 1 time
    4812098 (12.81%) aligned >1 times
15.97% overall alignment rate
CPU times: user 6.34 s, sys: 807 ms, total: 7.14 s
Wall time: 14min 16s


Проанализируйте выдачу bowtie. Почему процент выравниваний получился именно таким?

Имеет смысл для дальнейшего анализа отобрать уникально картированные риды.

Преобразование .sam в .bam (не требуется если запускать macs2 на .sam файлах)

In [ ]:
#!samtools view -h -S -b -o bowtie2_res/res_DXB.bam bowtie2_res/res_DXB.sam
#!samtools view -h -S -b -o bowtie2_res/res_KWQ.bam bowtie2_res/res_KWQ.sam
#!samtools view -h -S -b -o bowtie2_res/res_MPL.bam bowtie2_res/res_MPL.sam

На этом моменте можно скачать себе на компьютер все полученные .bam или .sam файлы на случай сбоя.

## Peak calling

Помимо macs2 может быть использован пакет Homer для поиска пиков, а также macs3.

In [ ]:
!mkdir macs2_DXB
!mkdir macs2_KWQ

mkdir: cannot create directory ‘macs2_DXB’: File exists
mkdir: cannot create directory ‘macs2_KWQ’: File exists


In [ ]:
!macs2 callpeak --broad -t bowtie2_res/res_DXB.sam \
    -c bowtie2_res/res_MPL.sam \
 	  -f SAM \
	  --outdir macs2_DXB

INFO  @ Wed, 09 Mar 2022 18:12:53: 
# Command line: callpeak --broad -t bowtie2_res/res_DXB.sam -c bowtie2_res/res_MPL.sam -f SAM --outdir macs2_DXB
# ARGUMENTS LIST:
# name = NA
# format = SAM
# ChIP-seq file = ['bowtie2_res/res_DXB.sam']
# control file = ['bowtie2_res/res_MPL.sam']
# effective genome size = 2.70e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff for narrow/strong regions = 5.00e-02
# qvalue cutoff for broad/weak regions = 1.00e-01
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 1000 bps and 10000 bps
# Broad region calling is on
# Paired-End mode is off
 
INFO  @ Wed, 09 Mar 2022 18:12:53: #1 read tag files... 
INFO  @ Wed, 09 Mar 2022 18:12:53: #1 read treatment tags... 
INFO  @ Wed, 09 Mar 2022 18:13:09: #1.2 read input tags... 
INFO  @ Wed

In [ ]:
!macs2 callpeak --broad -t bowtie2_res/res_KWQ.sam \
    -c bowtie2_res/res_MPL.sam \
 	  -f SAM \
	  --outdir macs2_KWQ

INFO  @ Wed, 09 Mar 2022 18:14:12: 
# Command line: callpeak --broad -t bowtie2_res/res_KWQ.sam -c bowtie2_res/res_MPL.sam -f SAM --outdir macs2_KWQ
# ARGUMENTS LIST:
# name = NA
# format = SAM
# ChIP-seq file = ['bowtie2_res/res_KWQ.sam']
# control file = ['bowtie2_res/res_MPL.sam']
# effective genome size = 2.70e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff for narrow/strong regions = 5.00e-02
# qvalue cutoff for broad/weak regions = 1.00e-01
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 1000 bps and 10000 bps
# Broad region calling is on
# Paired-End mode is off
 
INFO  @ Wed, 09 Mar 2022 18:14:12: #1 read tag files... 
INFO  @ Wed, 09 Mar 2022 18:14:12: #1 read treatment tags... 
INFO  @ Wed, 09 Mar 2022 18:14:29:  19000000 
INFO  @ Wed, 09 Mar 2022 

In [ ]:
!wget https://www.encodeproject.org/files/ENCFF090XFB/@@download/ENCFF090XFB.bed.gz
!gzip -d ENCFF090XFB.bed.gz

--2022-03-09 17:48:50--  https://www.encodeproject.org/files/ENCFF090XFB/@@download/ENCFF090XFB.bed.gz
Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144
Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.
HTTP request sent, awaiting response... 307 Temporary Redirect
Location: https://encode-public.s3.amazonaws.com/2017/01/31/2854b499-9022-425b-b163-7866f22da6d2/ENCFF090XFB.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF090XFB.bed.gz&AWSAccessKeyId=ASIATGZNGCNX4XLA2UVH&Signature=ZEmcubUvFC7F2RXBNXz8U7oZE3o%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEHIaCXVzLXdlc3QtMiJGMEQCIGGuz9teVbSH5o%2FgzS7vd5fFAjycUzc%2BhYmH3p2fxva7AiBaiMZJMAYDXKNpW%2FSoqOXNLjSuGzL8THCiFoHPSPXr5iqDBAjb%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMMHG8R%2BNTw5JRlMImKtcDl50sw80Bcn2tK3kCMHiUob2VZrW28GkIC7PQIDcPkg2NihtwqOCgNyv7GZk3Cqa3cdgnpZAIknSqUuHfW8bBYGgkMIIjUQsBaNMKX4C39y3x1rrRhhNlNsVocpsy4aTIZeiS6hGPEvUPgv8pG3%2BQOCEew0Z

## Сравнение результатов

Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.

Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?

In [ ]:
!intervene venn -i macs2_DXB/NA_peaks.broadPeak ENCFF090XFB.bed --filenames --output venn_results/venn1_DXB.jpg
!intervene venn -i ENCFF090XFB.bed macs2_DXB/NA_peaks.broadPeak --filenames --output venn_results/venn2_DXB.jpg


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ venn_results/venn1_DXB.jpg. 
Thank you for using Intervene!


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ venn_results/venn2_DXB.jpg. 
Thank you for using Intervene!



In [ ]:
!intervene venn -i macs2_KWQ/NA_peaks.broadPeak ENCFF090XFB.bed --filenames --output venn_results/venn1_KWQ.jpg
!intervene venn -i ENCFF090XFB.bed macs2_KWQ/NA_peaks.broadPeak --filenames --output venn_results/venn2_KWQ.jpg


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ venn_results/venn1_KWQ.jpg. 
Thank you for using Intervene!


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ venn_results/venn2_KWQ.jpg. 
Thank you for using Intervene!

